### DISCUSSION TITLE
How To Prevent Use of NotImplementedException

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui#313] [General]
Is there a way to use `editorconfig`, `Directory.build.props` or Roslyn Analyzers to tell the compiler to trigger a compilation error when `NotImplementedException` is detected?

In https://github.com/CommunityToolkit/Maui/pull/296 we replaced `NotImplementedException` with `NotSupportedException` because `NotImplementedException` implies that we will add support for a feature in the future.

Going forward, we'd like to avoid new PRs from adding code that throws `NotImplementedException`, however we cannot find an `editorconfig` rule that helps us accomplish this (https://github.com/CommunityToolkit/Maui/pull/296#issuecomment-1058431543).

**Comment by pictos** [comment]
Not sure about others, but with Roslyn Analyzers we can do it. I'll what I can do. If we came with an Analyzer would be good to create a repo in this organization to host it

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
